frontend/src/components/DefenceBox/PromptEnclosureDefenceMechanism.tsx

```diff
@@ -0,0 +1,106 @@
+import { useState } from 'react';
+
+import DefenceConfigurationRadioButton from '@src/components/ThemedButtons/DefenceConfigurationRadioButton';
+import { DEFENCE_ID, Defence } from '@src/models/defence';
+
+import DefenceConfiguration from './DefenceConfiguration';
+
+function PromptEnclosureDefenceMechanism({
+	defences,
+	toggleDefence,
+	showConfigurations,
+	setConfigurationValue,
+	resetConfigurationValue,
+}: {
+	defences: Defence[];
+	toggleDefence: (defence: Defence) => void;
+	showConfigurations: boolean;
+	setConfigurationValue: (
+		defence: Defence,
+		configId: string,
+		value: string
+	) => Promise<void>;
+	resetConfigurationValue: (defence: Defence, configId: string) => void;
+}) {
+	const [selectedRadio, setSelectedRadio] = useState<string>('none');
+	const selectedDefence = getDefence(selectedRadio as DEFENCE_ID);
+
+	function isRadioSelected(radioValue: string) {
+		return radioValue === selectedRadio;
+	}
+
+	function getDefence(defenceId: DEFENCE_ID): Defence | undefined {
+		// will be undefined if 'none' selected
+		return defences.find((defence) => defence.id === defenceId);
+	}
+
+	function deactivateAll() {
+		defences.forEach((defence) => {
+			if (defence.isActive) {
+				toggleDefence(defence);
+			}
+		});
+	}
+
+	// activate this defence and deactivate all others
+	function activateOne(defenceId: DEFENCE_ID) {
+		defences.forEach((defence) => {
+			if (defence.id === defenceId && !defence.isActive) toggleDefence(defence);
+			else if (defence.isActive) {
+				toggleDefence(defence);
+			}
+		});
+	}
+
+	function handleRadioChange(event: React.ChangeEvent<HTMLInputElement>) {
```

Comment:
I'm wondering if it might be better for this function to take just the string value, because the only time `event` is used here is to get the value using `event.target.value`. 

So this method would become
```typescript
function handleRadioChange(defenceId: string) {
```